src/components/Common/Button/index.tsx

```diff
@@ -8,10 +8,10 @@ export type ButtonType =
   | 'success'
   | 'ghost';
 
-// Helper type to override types (override's onClick)
+// Helper type to override types (overrides onClick)
 type MergeElementProps<
   T extends React.ElementType,
-  P extends Record<string, any>
+  P extends Record<string, unknown>
```

Comment:
With this change, does it still complain when you do this?:
```
  if (className) {
    buttonStyle.push(className);
  }
```
Just wondering, haha 😛

Reply:
Yeah, this doesn't affect the issue. Here it is if you're interested (inside the if-block). 
![image](https://user-images.githubusercontent.com/14110063/107005572-29d97200-6790-11eb-92e5-15a7f64d8178.png)

So the issue is that className sort of have the type `(string | undefined) | undefined`, if you understand what I mean.

Reply:
@ankarhem Oh huh.  I guess TypeScript can be temperamental sometimes 😛  Thanks for sharing!